static-assets/components/cstudio-common/common-api.js

```diff
@@ -2175,14 +2176,23 @@ var nodeOpen = false,
              * opens a dialog if needed or goes directly to the form if no
              * template selection is require (only one option
              */
-            createNewContent: function(site, path, asPopup, formSaveCb, childForm, isFlattenedInclude) {
+            createNewContent: function(site, path, asPopup, formSaveCb, childForm, isFlattenedInclude, filter) {
                 var auxParams = [];
                 if(childForm && childForm == true) {
                     auxParams = [ { name: "childForm", value: "true" }];
                 }
 
                 var callback = {
                     success: function(contentTypes) {
+                        if (filter) {
```

Comment:
Make this a function that would be sent to array.prototype.filter

Reply:
done

Reply:
This wasn’t the idea. The idea was that the consumer could send the filtering FUNCTION that you’d directly pass to array.filter. This way the filter is dynamic and not fixed to filtering the `.type` property.

Reply:
Done